language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/TemplateExpressionNodeContext.java

```diff
@@ -81,6 +90,36 @@ public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context,
         return completionItems;
     }
 
+    private List<LSCompletionItem> getRegexCompletions(NonTerminalNode nodeAtCursor, BallerinaCompletionContext ctx) {
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+        if (nodeAtCursor.kind() == SyntaxKind.RE_CHAR_ESCAPE 
+                && nodeAtCursor.toSourceCode().equals(SyntaxKind.BACK_SLASH_TOKEN.stringValue())) {
+            // Eg: re `\<cursor>`
+            completionItems.add(new SnippetCompletionItem(ctx, RegexSnippet.DEF_DIGIT.get()));
+            completionItems.add(new SnippetCompletionItem(ctx, RegexSnippet.DEF_NON_DIGIT.get()));
+            completionItems.add(new SnippetCompletionItem(ctx, RegexSnippet.DEF_WHITESPACE.get()));
+            completionItems.add(new SnippetCompletionItem(ctx, RegexSnippet.DEF_NON_WHITESPACE.get()));
+            completionItems.add(new SnippetCompletionItem(ctx, RegexSnippet.DEF_ALPHA_NUMERIC.get()));
+            completionItems.add(new SnippetCompletionItem(ctx, RegexSnippet.DEF_NON_ALPHA_NUMERIC.get()));
+            completionItems.add(new SnippetCompletionItem(ctx, RegexSnippet.DEF_RETURN.get()));
+            completionItems.add(new SnippetCompletionItem(ctx, RegexSnippet.DEF_NEWLINE.get()));
+            completionItems.add(new SnippetCompletionItem(ctx, RegexSnippet.DEF_TAB.get()));
+        } else if (nodeAtCursor.kind() == SyntaxKind.RE_FLAG_EXPR 
+                && ((ReFlagExpressionNode) nodeAtCursor).questionMark().position() + 1 == ctx
+                .getCursorPositionInTree()) {
```

Comment:
Shall we move this condition to a separate method with a proper name?

Reply:
Done